b2share/modules/schemas/views.py

```diff
@@ -139,12 +147,96 @@ def get(self, community_schema):
+    view_name = 'block_schema_list'
+
+    def __init__(self, **kwargs):
+        """Constructor."""
+        super(BlockSchemaListResource, self).__init__(
+            method_serializers={
+                'GET': {
+                    'application/json': schemas_list_to_json_serializer,
+                },
+                'POST': {
+                    'application/json': new_schema_to_json_serializer,
+                }
+            },
+            default_method_media_type={
+                'GET': 'application/json',
+                'POST': 'application/json',
+            },
+            default_media_type='application/json',
+            **kwargs
+        )
+
+    def get(self, community_id):
+        """Gets a list of all schemas."""
+        schemas = BlockSchema.get_all_block_schemas(community_id=community_id)
+        return self.make_response(
+            schemas=schemas,
+            code=200)
+
+    def post(self, community_id):
+        """Creates a new schema."""
+        data = request.get_json()
+        if data is None:
+            return abort(400)
+        schema = BlockSchema.create_block_schema(community_id, data['name'])
+        return self.make_response(
+            schema=schema,
+            code=201,
+        )
+
+
+class BlockSchemaResource(ContentNegotiatedMethodView):
+    view_name = 'block_schema_version'
+
+    def __init__(self, **kwargs):
+        """Constructor."""
+        super(BlockSchemaResource, self).__init__(
+            method_serializers={
+                'PUT': {
+                    'application/json': block_schema_version_to_json_serializer,
+                }
+            },
+            default_method_media_type={
+                'PUT': 'application/json'
+            },
+            default_media_type='application/json',
+            **kwargs
+        )
+
+    def put(self, schema_id):
+        """Creates a new version of the schema."""
+        data = request.get_json()
+        if data is None:
+            return abort(400)
+        block_schema = BlockSchema.get_block_schema(schema_id)
+        schema = block_schema.create_version(data['json_schema'])
+        return self.make_response(
+            block_schema_version=schema,
+            code=201,
+        )
+
+
 blueprint.add_url_rule(
     '/schemas/<string:schema_id>/versions/<int:schema_version_nb>',
     view_func=BlockSchemaVersionResource
-    .as_view(BlockSchemaVersionResource.view_name))
+        .as_view(BlockSchemaVersionResource.view_name))
+
 
 blueprint.add_url_rule(
     '/communities/<string:community_id>/schemas/<string:schema_version_nb>',
     view_func=CommunitySchemaResource
-    .as_view(CommunitySchemaResource.view_name))
+        .as_view(CommunitySchemaResource.view_name))
+
+
+blueprint.add_url_rule(
+    '/schemas/<string:schema_id>',
+    view_func=BlockSchemaResource
+        .as_view(BlockSchemaResource.view_name))
+
+blueprint.add_url_rule(
+    '/schemas/<string:community_id>',
```

Comment:
the endpoint should be `/schemas/.
For POST the community_id should be in the data.
for GET it should be given as a query parameter.